webapp/javascript/components/ProfilerTable.jsx

```diff
@@ -122,37 +209,93 @@ function TableBody({ flamebearer, sortBy, sortByDirection }) {
 
   const formatter = getFormatter(numTicks, sampleRate, units);
 
-  return sorted.map((x) => {
-    const pn = getPackageNameFromStackTrace(spyName, x.name);
-    const color = colorBasedOnPackageName(pn, 1);
-    const style = {
-      backgroundColor: color,
-    };
-    return (
-      <tr key={x.name+renderID}>
+  const renderRow =
+    !viewDiff ? (x, color, style) => (
+      <tr key={x.name + renderID}>
         <td>
           <span className="color-reference" style={style} />
           <span title={x.name}>{x.name}</span>
         </td>
         <td style={backgroundImageStyle(x.self, maxSelf, color)}>
           {/* <span>{ formatPercent(x.self / numTicks) }</span>
-        &nbsp;
-        <span>{ shortNumber(x.self) }</span>
-        &nbsp; */}
+      &nbsp;
+      <span>{ shortNumber(x.self) }</span>
+      &nbsp; */}
           <span title={formatter.format(x.self, sampleRate)}>
             {formatter.format(x.self, sampleRate)}
           </span>
         </td>
         <td style={backgroundImageStyle(x.total, numTicks, color)}>
           {/* <span>{ formatPercent(x.total / numTicks) }</span>
-        &nbsp;
-        <span>{ shortNumber(x.total) }</span>
-        &nbsp; */}
+      &nbsp;
+      <span>{ shortNumber(x.total) }</span>
+      &nbsp; */}
           <span title={formatter.format(x.total, sampleRate)}>
             {formatter.format(x.total, sampleRate)}
           </span>
         </td>
       </tr>
-    );
+    ) : viewDiff === 'self' ? (x, color, style) => (
+      <tr key={x.name + renderID}>
+        <td>
+          <span className="color-reference" style={style} />
+          <span title={x.name}>{x.name}</span>
+        </td>
+        {/* NOTE: it seems React does not understand multiple backgrounds, have to workaround:  */}
+        {/*   The `style` prop expects a mapping from style properties to values, not a string. */}
+        <td STYLE={backgroundImageDiffStyle(x.selfLeft, x.selfRght, maxSelf, color, 'L')}>
```

Comment:
Can you clarify this part a little more? I noticed the console has this error, but are you saying this is intended? 
![image](https://user-images.githubusercontent.com/23323466/127708856-be6522c7-3b10-4520-937d-4061093018fd.png)

Reply:
I try to use multiple backgrounds here. The original code that I tested (and it didn't work) was:

```html
// tried "style" instead of "STYLE"
<td style={backgroundImageDiffStyleALT(x.selfLeft, x.selfRght, maxSelf, color, 'L')}>
```
```js
function backgroundImageDiffStyleALT(a, b, total, color, side) {
  // ...
  // tried object instead of string for css
  return {
    backgroundImage: `linear-gradient(${clr}, ${clr}), linear-gradient(${cld}, ${cld})`;
    backgroundPosition: `${neg(-k)}px 0px, ${neg(-kd)}px 0px`;
    backgroundRepeat: `no-repeat`;
  };
}
```

It didn't work. And I had to change to the current code:

```html
// workaround: use "STYLE"
<td style={backgroundImageDiffStyle(x.selfLeft, x.selfRght, maxSelf, color, 'L')}>
```
```js
function backgroundImageDiffStyle(a, b, total, color, side) {
  // ...
  // workaround: use string for css
  return `
    background-image: linear-gradient(${clr}, ${clr}), linear-gradient(${cld}, ${cld});
    background-position: ${neg(-k)}px 0px, ${neg(-kd)}px 0px;
    background-repeat: no-repeat;
  `;
}
```

If you can make the original code work or find a better workaround, I'm happy to learn about it.

Reply:
I fixed the first warning `Warning: Each child in a list should have a unique "key" prop.`. My mistake.